### DISCUSSION TITLE
Leaderlease examples?

### DISCUSSION THREAD

**Discussion by r0bobo** [kube-rs/kube#462] [Q&A]
Are there any examples of implementing a leader lease to do safe rolling deploys of controllers?

(New to both Rust and kube-rs but loving it so far! It's a pleasure to use compared to what I'm used to with go.)

**Comment by nightkr** [comment]
Not really, it's an area I've been meaning to look into but even the official K8s documentation on the topic is.. lacklustre at best (or at least it was the last time I tried to make sense of it). 

In the meantime, you might be able to get close enough with a `StatefulSet`? Alternately, it might be viable to structure your controller in such a way that running multiple instances is viable (rollback on update conflicts, and so on).

**Reply by r0bobo** [reply]
I looked at the leaderlease code in client-go and it was more complex than I hoped it would be. But maybe I'll take a stab at doing something similar in Rust if I can grok it.

**Comment by nightkr** [comment]
Also moving this into a discussion, since this isn't really an actionable proposal.
